docs/install/osx_setup.md

```diff
@@ -89,25 +92,31 @@ Install the dependencies, required for MXNet, with the following commands:
 	# Get pip
 	easy_install pip
 	# For visualization of network graphs
-	pip install graphviz
+	pip3 install graphviz
 	# Jupyter notebook
-	pip install jupyter
+	pip3 install jupyter
 ```
 
 ### Build MXNet Shared Library
 After you have installed the dependencies, pull the MXNet source code from Git and build MXNet to produce an MXNet library called ```libmxnet.so```. You can clone the repository as described in the following code block, or you may try the <a href="download.html">download links</a> for your desired MXNet version.
 
-The file called ```osx.mk``` has the configuration required for building MXNet on OS X. First copy ```make/osx.mk``` into ```config.mk```, which is used by the ```make``` command:
-
 ```bash
-    git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
-    cd ~/mxnet
-    cp make/osx.mk ./config.mk
-    echo "USE_BLAS = openblas" >> ./config.mk
-    echo "ADD_CFLAGS += -I/usr/local/opt/openblas/include" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/opt/openblas/lib" >> ./config.mk
-    echo "ADD_LDFLAGS += -L/usr/local/lib/graphviz/" >> ./config.mk
-    make -j$(sysctl -n hw.ncpu)
+git clone --recursive https://github.com/apache/incubator-mxnet ~/mxnet
+cd ~/mxnet && pushd .
+mkdir build && cd build
+cmake \
+    -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+    -DCMAKE_C_COMPILER_LAUNCHER=ccache \
+    -DUSE_MKL_IF_AVAILABLE=OFF \
```

Comment:
Can we make this single line. Comparing to the Makefile instruction, this is too complicated.

Reply:
the lines are escaped, so it can be copied and pasted no problem.

Reply:
<img width="610" alt="Screen Shot 2019-06-05 at 3 47 53 PM" src="https://user-images.githubusercontent.com/928489/58995546-8ad5c900-87a9-11e9-8b7f-e11c74fa5180.png">

Reply:
@pengzhao-intel are these the optimal compile flags? Especially with regard to omp and mkl?

Also, why are we so verbose here? Are our defaults not good enough?

Reply:
A build process should be as simple as possible. Ideally, after a few times a user can build it without having to find this page and do copy-paste. I've found it difficult to remember all these flags and it's really frustrating to developers to open this page every time when they want to build mxnet from source.